app/services/cohort_service.rb

```diff
@@ -1,36 +1,37 @@
 class CohortService
-  attr_reader :region, :quarters
   CACHE_VERSION = 1
   CACHE_TTL = 7.days
+  attr_reader :range
+  attr_reader :region
 
-  def initialize(region:, quarters: nil)
+  def initialize(region:, range: nil)
     @region = region
-    @quarters = quarters || default_quarters
+    @range = range || default_range
   end
 
   # Each quarter cohort is made up of patients registered in the previous quarter
   # who has had a follow up visit in the current quarter.
   def call
     result = {quarterly_registrations: []}
-    quarters.each do |results_quarter|
-      quarter_data = compute_quarter(results_quarter)
-      result[:quarterly_registrations] << quarter_data
+    range.each do |period|
+      result[:quarterly_registrations] << compute(period)
     end
     result
   end
 
   private
 
-  def compute_quarter(results_quarter)
-    Rails.cache.fetch(cache_key(results_quarter), version: cache_version, expires_in: CACHE_TTL, force: force_cache?) do
-      cohort_quarter = results_quarter.previous_quarter
-      period = {cohort_period: :quarter,
-                registration_quarter: cohort_quarter.number,
-                registration_year: cohort_quarter.year}
+  def compute(period)
+    Rails.cache.fetch(cache_key(period), version: cache_version, expires_in: CACHE_TTL, force: force_cache?) do
+      cohort_period = period.previous
+      period = {cohort_period: cohort_period.type,
+                registration_quarter: cohort_period.value.try(:number),
+                registration_year: cohort_period.value.try(:year),
+                registration_month: cohort_period.value.try(:month)}
```

Comment:
I guess we don't _need_ to, but should we version-bump the cache for good measure?